src/librustc_privacy/lib.rs

```diff
@@ -347,19 +341,41 @@ impl<'a, 'tcx, 'v> Visitor<'v> for EmbargoVisitor<'a, 'tcx> {
                 }
             }
 
+            hir::ItemForeignMod(ref foreign_mod) => {
+                for foreign_item in &foreign_mod.items {
+                    let public = self.prev_public && foreign_item.vis == hir::Public;
+                    let exported = self.prev_exported && foreign_item.vis == hir::Public ||
```

Comment:
Could you add parens around one of the clauses here? I always forget the precedence of `&&` and `||`...